src/linear_algebra/matrix/charpoly/coeff.lean

```diff
@@ -190,3 +190,43 @@ lemma pow_eq_aeval_mod_charpoly (M : matrix n n R) (k : ℕ) : M^k = aeval M (X^
 by rw [←aeval_eq_aeval_mod_charpoly, map_pow, aeval_X]
 
 end matrix
+
+section ideal
+
+lemma coeff_prod_mem_ideal_pow_tsub {α : Type*} (s : finset α) (f : α → R[X])
+  (I : ideal R) (n : α → ℕ) (h : ∀ i k, (f i).coeff k ∈ I ^ (n i - k)) (k : ℕ) :
```

Comment:
`h` is needed only for `i` in `s`, right? Also can you avoid using `k` twice (even if logically correct it makes difficult to read the statement) and add a docstring? Also, this lemma should be moved to a more basic file.

Reply:
I'm not sure how to avoid the two `k`'s in `h`? Or do you mean that I should rename the `k` in `(s.prod f).coeff k ∈ I ^ (s.sum n - k)`?

Reply:
I just mean to use a different name for the last `(k : ℕ)`, to avoid having to `k`'s in the same line with different meaning (of course the first one is after `∀`, so it technically OK, but a little confusing).

Reply:
Or use another name for the first one, as you wish.

Reply:
Would it be better if I put `(k : ℕ)` in the next line after the colon?
My thought is that it would be better to make
`∀ k, (f i).coeff k ∈ I ^ (n i - k)` and
`∀ k, (s.prod f).coeff k ∈ I ^ (s.sum n - k)` look alike.

Reply:
Hmm, you're maybe right, let's keep it as it is, it's not so important.